### PR TITLE
#14974: Clean up creation functions in the ttnn numpy package

### PR DESCRIPTION
**Pull request by omilyutin-tt**
### Ticket
#14974 

### Problem description
Creation functions like `zeros`, `ones`, `full` have a dedicated place `creation.hpp`. Currently, several unit tests rely on the redundant interfaces defined in `functions.hpp` in the numpy package.

### What's changed
* Migrated the use of `numpy` versions of `zeros`, `ones`, `full`, `arange` in tests to `ttnn::{zeros,ones,full}`.
* Moved the base implementation of `full` and `arange` into `creation.hpp` - the only place where they are used.
* Minor simplification / refactor changes to deduplicate code.

### Checklist
- [X] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/12152247431)
- [X] [T3K unit + frequent tests pass](https://github.com/tenstorrent/tt-metal/actions/runs/12152254447)
- [X] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/creation.hpp [hunk 1]
```diff
@@ -418,12 +456,11 @@ struct FullLike {
     }
 };
 
-// TODO: #14974 - Onboard this API onto AnyDevice.
 struct Arange {
     static ttnn::Tensor invoke(
         const int64_t stop,
         const DataType dtype = DataType::BFLOAT16,
-        const std::optional<std::reference_wrapper<Device>>& device = std::nullopt,
+        detail::OptionalAnyDevice device = std::nullopt,
```
Comment (cfjchu) [inline ttnn/cpp/ttnn/operations/creation.hpp:463]: do we have meshdevice coverage here?

Comment (omilyutin-tt) [reply ttnn/cpp/ttnn/operations/creation.hpp:463]: We didn't - addressed.

**Review by dmakoviichuk-tt — COMMENTED** [review]
I'd remove numpy name from the ttnn at all. It is a little bit missleading.

**Comment by omilyutin-tt** [comment]
@dmakoviichuk-tt @ayerofieiev-tt agree, let's do in a follow up: move the header out of `numpy/`, and get rid of the namespace?

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by cfjchu — APPROVED** [review]
Thanks for adding the unit tests!

**Review by tt-asaigal — APPROVED** [review]

**Pull request merged**